tests/api2/test_audit_basic.py

```diff
@@ -15,10 +14,60 @@
 
 SMBUSER = 'audit-smb-user'
 PASSWD = ''.join(secrets.choice(string.ascii_letters + string.digits) for i in range(10))
+AUDIT_DATASET_CONFIG = {
+    # keyname : "audit"=audit only setting, "zfs"=zfs dataset setting, "ro"=read-only (not a setting)
+    'retention': 'audit',
+    'quota': 'zfs',
+    'reservation': 'zfs',
+    'quota_fill_warning': 'zfs',
+    'quota_fill_critical': 'zfs',
+    'remote_logging_enabled': 'other',
+    'space': 'ro'
+}
+MiB = 1024**2
+GiB = 1024**3
+
+
+# =====================================================================
+#                     Fixtures and utilities
+# =====================================================================
+class AUDIT_CONFIG():
+    defaults = {
+        'retention': 7,
+        'quota': 0,
+        'reservation': 0,
+        'quota_fill_warning': 75,
+        'quota_fill_critical': 95
+    }
+
+
+# def get_zfs(key, zfs_config):
+def get_zfs(type, key, zfs_config):
+    """ Get the equivalent ZFS value associated with the audit config setting """
+
+    types = {
+        'zfs': {
+            'reservation': zfs_config['properties']['refreservation']['parsed'] or 0,
+            'quota': zfs_config['properties']['refquota']['parsed'] or 0,  # audit quota == ZFS refquota
+            'refquota': zfs_config['properties']['refquota']['parsed'] or 0,
+            'quota_fill_warning': zfs_config['org.freenas:quota_warning'],
+            'quota_fill_critical': zfs_config['org.freenas:quota_critical']
+        },
+        'space': {
+            'used': zfs_config['properties']['used']['parsed'],
+            'used_by_snapshots': zfs_config['properties']['usedbysnapshots']['parsed'],
+            'available': zfs_config['properties']['available']['parsed'],
+            'used_by_dataset': zfs_config['properties']['usedbydataset']['parsed'],
+            # We set 'refreservation' and there is no 'usedbyreservation'
+            'used_by_reservation': zfs_config['properties']['usedbyrefreservation']['parsed']
+        }
+    }
+    # return zfs[key]
+    return types[type][key]
 
 
-@pytest.fixture(scope='module')
-def initialize_for_smb_tests(request):
+@pytest.fixture(scope='class')
```

Comment:
Why are we changing scope?

Reply:
Only need the smb stuff for the operations section.